### PR TITLE
Moving getEncodedFileKey to common function and fixing unit tests

### LINKED ISSUE #3409: No universal method for getting EncodedFileKey
**Is your feature request related to a problem? Please describe.**
The fileKeyEncoding is done in multiple places in the Civiform codebase. We need a universal method that the clients can use. 

**Describe the solution you'd like**
Create a helper method getEncodedFileKey() in AnswerData so that PDFExporter, ProgramApplicationView and other clients can it.

### PR DESCRIPTION
**Pull request by swatkat1**
### Description

Moved EncodedFileKet as a memeber of the AnswerData class. It will be a common field like the FileKey(). 

## Release notes:

EncodedFileKey is moved a universal location. So all file downloads should be using it.

### Checklist

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

Fixes #3409

### CODE REVIEW

File: server/app/services/applicant/AnswerData.java [hunk 1]
```diff
@@ -53,6 +53,8 @@ public static Builder builder() {
   /** The identifier of the applicant's uploaded file if applicable. */
   public abstract Optional<String> fileKey();
```
Comment (clouser2) [inline server/app/services/applicant/AnswerData.java:54]: I believe there won't be any remaining callers of this, so it can be removed.

File: server/app/services/applicant/AnswerData.java [hunk 2]
```diff
@@ -53,6 +53,8 @@ public static Builder builder() {
...
 
+  public abstract Optional<String> encodedFileKey();
```
Comment (clouser2) [inline server/app/services/applicant/AnswerData.java:56]: Please add a docstring.

File: server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java [hunk 3]
```diff
@@ -186,10 +186,14 @@ public ImmutableList<AnswerData> getSummaryData() {
         Optional<Long> updatedProgram = question.getUpdatedInProgramMetadata();
         Optional<String> originalFileName = Optional.empty();
         Optional<String> fileKey = Optional.empty();
+        Optional<String> encodedFileKey = Optional.empty();
         if (isAnswered && question.isFileUploadQuestion()) {
           FileUploadQuestion fileUploadQuestion = question.createFileUploadQuestion();
           originalFileName = fileUploadQuestion.getOriginalFileName();
           fileKey = fileUploadQuestion.getFileKeyValue();
```
Comment (clouser2) [inline server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java:193]: I think the fileKey variable can be inlined if we no longer need to call `setFileKey()` below.

E.g:
```
Optional<String> originalFileName = Optional.empty();
Optional<String> encodedFileKey = Optional.empty();
if (isAnswered && question.isFileUploadQuestion()) {
  FileUploadQuestion fileUploadQuestion = question.createFileUploadQuestion();
  originalFileName = fileUploadQuestion.getOriginalFileName();
  encodedFileKey = fileUploadQuestion.getFileKeyValue().map((fileKey) -> URLEncoder.encode(fileKey, StandardCharsets.UTF_8));
}
```

**Review by clouser2 — CHANGES_REQUESTED** [review]

File: server/test/views/BaseHtmlLayoutTest.java [hunk 4]
```diff
@@ -26,7 +26,8 @@ public void addsDefaultContent() {
     assertThat(content.body()).contains("<!DOCTYPE html><html lang=\"en\">");
 
     assertThat(content.body())
-        .containsPattern("<link href=\"/assets/stylesheets/tailwind.css\" rel=\"stylesheet\">");
+        .containsPattern(
+            "<link href=\"/assets/stylesheets/[a-z0-9]+-tailwind.css\" rel=\"stylesheet\">");
```
Comment (clouser2) [inline server/test/views/BaseHtmlLayoutTest.java:30]: Out of curiosity, how are the changes to this file relevant? They seem unrelated.

**Review by clouser2 — APPROVED** [review]

File: server/test/views/BaseHtmlLayoutTest.java (hunk 4 above)
Comment (swatkat1) [reply server/test/views/BaseHtmlLayoutTest.java:30]: Without this the unit tests were failing..

Comment (nbeloglazov) [reply server/test/views/BaseHtmlLayoutTest.java]: That's interesting. That seems to be related to my change of enabling fingerprinting for all files. I submitted it earlier this week. And I did expect tailwind.css to have fingerprint as well but for some reason the test didn't fail so I didn't update it. Hm, you reverted this change because it's no longer failing?

Comment (swatkat1) [reply server/test/views/BaseHtmlLayoutTest.java]: Yes , it worked fine in CI even after reverting it.. so reverted it.
